File: AlphaBack/UniBackEnd/src/main/java/com/esprit/unibackend/entities/Reclamation.java
package com.esprit.unibackend.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Reclamation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;

    private String description;
    private String category;
    private Date date;

    @Enumerated(EnumType.STRING)
    private StatusRec status;

    @ManyToOne
    private User creator;
}

File: AlphaFront/src/app/models/Reclamation.ts
import { User } from "./User";
export enum StatusRec{
    EnAttente,Valide,Refuse
}
export class Reclamation{
    id!:number;
    description!:string;
    date!:Date;
    creator:User  |undefined;
    category!:string;
    status!:StatusRec;
}